public/main.js

```diff
@@ -0,0 +1,43 @@
+// Variable Declarations ////////////////////////////////////////////////////////////////////////
+let myMap = (mapboxgl.accessToken =
+  "pk.eyJ1Ijoic2Vhbm1wMjIiLCJhIjoiY2tjM3M0bGlwMDB4ejJxbXNmbndvdjNxbyJ9.vwuXpXD9aoHqCUtonVTF0A");
+let map = new mapboxgl.Map({
+  container: "map", // container id
+  style: "mapbox://styles/mapbox/streets-v11", // stylesheet location
+  center: [-73.2150, 44.4810], // starting position [lng, lat]
+  zoom: 15, // starting zoom
+});
+let restoNameArray = [];
+let restoCoords = [];
+let restaurantList = document.getElementById('restoList')
+
+// Function Declarations ////////////////////////////////////
+function placeMarker(coords) {  // Place marker on map
+       let lngLatArr = [coords[1], coords[0]];
+       new mapboxgl.Marker().setLngLat(lngLatArr).addTo(map);
+    };
+
+
+async function restaurantsOnMap() {  // Fetch API data and push relavent data to arrays for later use
+    let restaurantList = await fetch('https://yelpingtonapi.herokuapp.com/api/restaurants')
+    .then((res) => res.json())
+    
+    
+    for (restaurant of restaurantList) {
+        restoNameArray.push(restaurant.name)
+        restoCoords.push(restaurant.coords)
+    }
+    restoCoords.forEach(coords => {
+        placeMarker(coords) // Loop to place Lng/Lat markers on map
+    })
+    restoNameArray.forEach(restaurant => { // Loop to fill Nav Bar with restaurant name data
+      let node = document.createElement('LI')
+      let textNode = document.createTextNode(restaurant)
+      node.appendChild(textNode)
+      document.getElementById('restoList').appendChild(node);
+      })
+}
+
```

Comment:
Really awesome start, but I would argue the arrays are unnecessary, especially given the information you've pushed to them! Once you fetch `restaruantList`, that list contains all of the information you need to both generate the home page (like you have) AND the list of links in the restaurant list. Each object in `restaurantList` contains an `id`,  and as evidenced by their `infoUrl` properties, is all you need to to add to your NEXT `fetch` to get the individual restaurant data.

That individual restaurant data will exist eventually on your `restaurants.html` page (or whatever you choose to call it), and will dictate the specific end point you build that page with.  Just like you did on the home page with `restaurantList`!